sweet-exp-lib/sweet-exp/main.rkt

```diff
@@ -68,9 +30,7 @@
   #:info get-info
   #:language-info #((submod sweet-exp language-info) get-language-info #f)
 
-  (require (submod ".." link-reader))
-  (define-values (sweet-read sweet-read-syntax)
-    (sweet-link read read-syntax))
```

Comment:
So I think originally the intention here was that by calling `sweet-link` here, it would link in the reader of the language that is the base of the mixin (the `<lang>` in `#lang sweet-exp <lang>`), to be able to support non-standard Racket datum syntax.

I'm not sure if this actually worked well in practice though, I see that for example it didn't seem to work correctly for Typed Racket reader syntax.

But this PR seems to always tie the linked reader to the racket/base reader, so I think it's a change in behavior. Does that makes sense or am I missing how this should work?

Reply:
Preserving the behavior of the `<lang>` reader in `#lang sweet-exp <lang>` never worked, as far as I'm aware. The first two lines of this module:
```racket
(module reader syntax/module-reader
  #:language read
  ...)
```
treat the `<lang>` in `#lang sweet-exp <lang>` as a module-language only, not as a reader language.

Reply:
Ah I see. I guess we should just keep the existing behavior then and merge this.

It's not needed for this PR, but is it actually possible to do the intended behavior where it treats the underlying language as a reader language?